File: src/dataset/javascript/space-age_exercism-javascript-ThomasZumsteg/openAI/ChatGPT4_space-age_v3.js
function SpaceAge(seconds) {
    this.seconds = seconds;
    this.earthYear = 31557600; // seconds
}

SpaceAge.prototype.calculateAge = function (ratio) {
    return parseFloat((this.seconds / (this.earthYear * ratio)).toFixed(2));
};

SpaceAge.prototype.onMercury = function () {
    return this.calculateAge(0.2408467);
};
SpaceAge.prototype.onVenus = function () {
    return this.calculateAge(0.61519726);
};
SpaceAge.prototype.onEarth = function () {
    return this.calculateAge(1.0);
};
SpaceAge.prototype.onMars = function () {
    return this.calculateAge(1.8808158);
};
SpaceAge.prototype.onJupiter = function () {
    return this.calculateAge(11.862615);
};
SpaceAge.prototype.onSaturn = function () {
    return this.calculateAge(29.447498);
};
SpaceAge.prototype.onUranus = function () {
    return this.calculateAge(84.016846);
};
SpaceAge.prototype.onNeptune = function () {
    return this.calculateAge(164.79132);
};

export default SpaceAge;

File: src/dataset/Java/space-age_exercism-java-blogscot/openAI/ChatGPT4_space-age_v3.java
import java.util.EnumMap;
import java.util.Map;

class SpaceAge {
  private final double seconds;

  private static final double EARTH_PERIOD = 31557600.0;

  private enum Planet {
    EARTH(1.0),
    MERCURY(0.2408467),
    VENUS(0.61519726),
    MARS(1.8808158),
    JUPITER(11.862615),
    SATURN(29.447498),
    URANUS(84.016846),
    NEPTUNE(164.79132);

    private final double ratio;

    Planet(double ratio) {
      this.ratio = ratio;
    }

    double getRatio() {
      return ratio;
    }
  }

  private final Map<Planet, Double> cachedAges = new EnumMap<>(Planet.class);

  SpaceAge(double seconds) {
    this.seconds = seconds;
  }

  private double calculateAge(Planet planet) {
    return cachedAges.computeIfAbsent(planet, p -> seconds / EARTH_PERIOD / p.getRatio());
  }

  double onEarth() {
    return calculateAge(Planet.EARTH);
  }

  double onMercury() {
    return calculateAge(Planet.MERCURY);
  }

  double onVenus() {
    return calculateAge(Planet.VENUS);
  }

  double onMars() {
    return calculateAge(Planet.MARS);
  }

  double onJupiter() {
    return calculateAge(Planet.JUPITER);
  }

  double onSaturn() {
    return calculateAge(Planet.SATURN);
  }

  double onUranus() {
    return calculateAge(Planet.URANUS);
  }

  double onNeptune() {
    return calculateAge(Planet.NEPTUNE);
  }
}